pkg/utils/kubectl.go

```diff
@@ -89,16 +93,39 @@ func CheckAllCommands(kubeconfigPath string, isContextSet bool, contextName stri
 	return nil
 }
 
-// checkAuthenticator checks for the authenticator binary existence.
+const (
+	authenticatorCommand       = "aws-iam-authenticator"
+	legacyAuthenticatorCommand = "heptio-authenticator-aws"
+)
+
+var authenticatorCommands = []string{
+	authenticatorCommand,
+	legacyAuthenticatorCommand,
+}
+
+// checkAuthenticator checks if either of authenticator commands is in the path,
+// it returns an error when neither are found
 func checkAuthenticator() error {
-	binaries := []string{"heptio-authenticator-aws", "aws-iam-authenticator"}
-	for _, bin := range binaries {
-		path, err := exec.LookPath(bin)
+	for _, cmd := range authenticatorCommands {
+		path, err := exec.LookPath(cmd)
 		if err == nil {
-			// binary was found
-			logger.Debug("%s: %q", bin, path)
+			// command was found
+			logger.Debug("%s: %q", cmd, path)
 			return nil
 		}
 	}
 	return fmt.Errorf("neither aws-iam-authenticator nor heptio-authenticator-aws are installed")
```

Comment:
Feel free to ignore this one! Would be good if the error message used the constants defined so there is less chance of deviation of error message from the commands.

Reply:
Yes, I thought about this, but we will soon care less about the legacy command, so I decided not too worry.